.github/workflows/model_ci.yml

```diff
@@ -42,3 +42,119 @@ jobs:
+    - uses: actions/checkout@v1
+    - name: GCP Auth
+      env:
+        GCLOUD_SERVICE_ACCOUNT_KEYFILE: ${{ secrets.GCLOUD_AUTH }}
+      run: |
+        echo "${GCLOUD_SERVICE_ACCOUNT_KEYFILE}" | base64 -d  > gcloud_sa_key.json
+        gcloud auth activate-service-account --key-file=gcloud_sa_key.json
+    - name: Install Chart
+      env:
+        DB_PASSWORD: ${{ secrets.DB_PASSWORD }}
+        RASA_TOKEN: ${{ secrets.RASA_TOKEN }}
+        RASA_X_TOKEN: ${{ secrets.RASA_X_TOKEN }}
+        JWT_SECRET: ${{ secrets.JWT_SECRET }}
+        PASSWORD_SALT: ${{ secrets.PASSWORD_SALT }}
+        REDIS_PASSWORD: ${{ secrets.REDIS_PASSWORD }}
+        RABBITMQ_PASSWORD: ${{ secrets.RABBITMQ_PASSWORD }}
+        RASA_X_ADMIN_PASSWORD: ${{ secrets.RASA_X_ADMIN_PASSWORD }}
+      run: |
+        # Install helm v3
+        curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash
+
+        gcloud container clusters get-credentials rasa-bots --zone europe-west1 --project rasa-platform
+        helm repo add rasa-x https://rasahq.github.io/rasa-x-helm
+
+        # Read and export variables from .env file
+        set -o allexport; source .env; set +o allexport
+
+        helm upgrade --install carbon-assistant rasa-x/rasa-x --namespace carbon-assistant \
+            --set global.postgresql.postgresqlPassword="${DB_PASSWORD}" \
+            --set app.tag=$(basename ${{ github.ref }})-${{ github.sha }} \
+            --set rasax.initialUser.password="${RASA_X_ADMIN_PASSWORD}" \
+            --set rasax.token=${RASA_TOKEN} \
+            --set rasax.jwtSecret=${JWT_SECRET} \
+            --set rasa.token=${RASA_X_TOKEN} \
+            --set rasax.passwordSalt=${PASSWORD_SALT} \
+            --set global.redis.password=${REDIS_PASSWORD} \
+            --set rabbitmq.rabbitmq.password=${RABBITMQ_PASSWORD} \
+            --set rasax.disableTelemetry=True \
+            --set rasax.tag="$RASA_X_VERSION" \
+            --set eventService.tag="${RASA_X_VERSION}" \
+            --set nginx.tag="${RASA_X_VERSION}" \
+            --set rasa.tag="${RASA_VERSION}-full" \
+            --values helm-values.yml
+
+        cat <<EOF | kubectl apply --namespace carbon-assistant -f  -
+        apiVersion: networking.gke.io/v1beta1
+        kind: ManagedCertificate
+        metadata:
+          generation: 1
+          name: rasa-bots-certificate
+        spec:
+          domains:
+          - carbon.rasa-bots.com
+        EOF
+
+  upload-model:
+    name: Upload the trained model to Rasa X
+    needs: 
+    - deploy-to-cluster
+    - build-model
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Model
+      uses: actions/download-artifact@master
+      with:
+        name: model
+    - name: Upload Model to Rasa
+      env:
+        RASA_X_PASSWORD: ${{ secrets.RASA_X_ADMIN_PASSWORD }}
+        RASA_X_USERNAME: admin
+      run: |
+        sleep 240
```

Comment:
One thing for the future: I think it would be great to get rid of statements like `sleep 240` and replace with something more reliable like checks while loop with checks if the deployment is ready.